src/services/tx/tx-sender/sdk.ts

```diff
@@ -29,3 +33,34 @@ export const getUncheckedSafeSDK = (provider: Web3Provider): Promise<Safe> => {
 
   return sdk.connect({ ethAdapter })
 }
+
+type SigningMethods = Parameters<Safe['signTransaction']>[1]
+
+export const _getSupportedSigningMethods = (safeVersion: SafeInfo['version']): SigningMethods[] => {
+  const SAFE_VERSION_SUPPORTS_ETH_SIGN = '>=1.1.0'
+
+  if (!safeVersion || !semverSatisfies(safeVersion, SAFE_VERSION_SUPPORTS_ETH_SIGN)) {
+    return ['eth_signTypedData']
+  }
+
+  return ['eth_signTypedData', 'eth_sign']
+}
+
+export const tryOffChainSigning = async (
+  safeTx: SafeTransaction,
+  safeVersion: SafeInfo['version'],
+): Promise<SafeTransaction | undefined> => {
```

Comment:
I don't think it can return undefined, can it? It either throws or returns.

Reply:
I think so, though it's not intelligent enough to pick it up unless we explicitly `throw` again at the end of the function.